### PR TITLE
Add active_line_width setting to increase the thickness of the active indent guide

### LINKED ISSUE #12312: Treesitter scope highlighting
### Check for existing issues

- [X] Completed

### Describe the feature

As we now have indent guides (#11503), we can now implement an incredibly useful feature which I've only seen in neovim. It highlights the current scope by making a guide thicker (see screenshots for details)

### If applicable, add mockups / screenshots to help present your vision of the feature

Screenshot 2024-05-26 at 13 29 19

### PR DESCRIPTION
**Pull request by WeetHet**
Resolves #12312.

Release Notes:

- Added an option to configure the line width of the active indent guide [#12312](https://github.com/zed-industries/zed/issues/12312)

### CODE REVIEW

**Comment by amtoaer** [comment]
I think there is no need to use `default_false` because boolean type defaults to `false`, `#[serde(default)]` might be enough.

File: crates/language/src/language_settings.rs [hunk 1]
```diff
@@ -443,6 +443,11 @@ pub struct IndentGuideSettings {
     /// Default: 1
     #[serde(default = "line_width")]
     pub line_width: u32,
+    /// Whether to make an active indent guide thicker than the others.
+    ///
+    /// Default: false
+    #[serde(default)]
+    pub thicken_active: bool,
```
Comment (SomeoneToIgnore) [inline crates/language/src/language_settings.rs:450]: Let's
* keep this on by default
* update `"indent_guides"` section inside `assets/settings/default.json` with the new default and docs

**Review by SomeoneToIgnore — COMMENTED** [review]
Interesting, looks like `is_active` does not consider nested tab symbols, is neovim doing the same?

image

Otherwise, looks good but let's enable this by default.

**Comment by WeetHet** [comment]
> Interesting, looks like `is_active` does not consider nested tab symbols, is neovim doing the same?
> 
> image
> 
> Otherwise, looks good but let's enable this by default.

Screenshot 2024-06-20 at 14 52 37

Seems like it doesn't. It also renders only one guide deeper than the active

**Comment by WeetHet** [comment]
> I think it makes more sense to add a setting for completely overriding the width of the active line (something like `active_line_width`), instead of always adding one. That way everyone can choose how thick/thin the active indent guide should be.

Oh yeah, that makes so much sense

**Comment by bennetbo** [comment]
Also, I do not think people expect this as the default, so let's set `active_line_width` to `1´ by default. Would be nice to add an entry in the [docs](https://github.com/zed-industries/zed/blob/main/docs/src/configuring-zed.md#indent-guides) as well

**Comment by SomeoneToIgnore** [comment]
Why not, though?
Highlighting the current block aligns with the rest of our "current match", "current symbol", etc., so I'd love to see this as a default too, as it's not very discoverable otherwise.

**Comment by WeetHet** [comment]
Huh, why did the tests fail to pass

**Comment by bennetbo** [comment]
> Why not, though? Highlighting the current block aligns with the rest of our "current line", "current match", "current symbol", etc. highlights, so I'd love to see this as a default too, as it's not very discoverable otherwise.

We do highlight them, yes, but we usually do not change the size of the text/lines/etc. only the color (as far as I am aware). From a personal standpoint I like the subtle coloring for active indent guides and find different line width for them distracting (this is of course highly subjective).
From a more objective-standpoint, I have not seen this "thickening" behavior used as the default in any of the editors I've used (VSCode, Visual Studio, multiple Jetbrains products, Sublime)

File: crates/editor/src/element.rs [hunk 2]
```diff
@@ -2782,6 +2782,11 @@ impl EditorElement {
             };
 
             let requested_line_width = settings.line_width.clamp(1, 10);
+            let requested_line_width = if indent_guide.active {
```
Comment (bennetbo) [inline crates/editor/src/element.rs:2785]: small NIT: We could probably merge this with the line above, something like:

```rs
if indent_guide.active {
    settings.active_line_width
} else {
    settings.line_width
}
.clamp(1, 10);
```

**Comment by bennetbo** [comment]
Let's add an entry to the [docs](https://github.com/zed-industries/zed/blob/main/docs/src/configuring-zed.md#indent-guides), and then this should be good to go

**Comment by bennetbo** [comment]
Thanks!

**Pull request merged**